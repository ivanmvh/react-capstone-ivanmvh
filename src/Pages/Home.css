#homeContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

header {
  height: 20vh;
  display: flex;
  width: 100vw;
}

#headerImage {
  flex-basis: 55%;
  height: 100%;
  background: url(../Resources/Images/header2.jpg);
  background-size: cover;

  /* background-position-x: left;
  background-position-y: top; */
}

#headerText {
  flex-basis: 45%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  font-size: 0.5rem;
  background-color: rgba(0, 0, 0, 0.847);
  color: white;
}

#headerDescription {
  width: 90%;
  margin-top: 12px;
  font-style: italic;
  font-size: 0.8rem;
}

#discoverAction {
  display: none;
}

#gridTitle {
  font-size: 1rem;
  width: 100%;
  height: 2rem;
  border: 1px solid rgb(177, 57, 103);
  padding-left: 5px;
  display: flex;
  align-items: center;
}

#desktopGridTitle {
  display: none;
}

hr {
  display: none;
}

#categoryGrid {
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.categoryContainer {
  height: 30vh;
  border: none;
  display: flex;
  align-items: flex-end;
  padding: 20px;
  background-color: rgb(177, 57, 103);
}

.categoryContainer:nth-child(4n+1) {
  background-color: rgb(236, 76, 138);
}

.categoryContainer:nth-child(4n+4) {
  background-color: rgb(236, 76, 138);
}

.categoryLink {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  background-color: white;
  text-decoration: none;
  color: rgb(255, 255, 255);
  font-weight: bold;
  font-size: 17px;
}

.categoryTitle {
  padding-right: 5%;
  text-decoration: none;
  width: 100%;
  height: 50px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  background-color: rgb(177, 57, 103, 0.237);
}

.categoryTitle:hover {
  background-color: rgba(0, 0, 0, 0.719);
}

/* styles categories */

.backgrounds {
  background: url(../Resources/Images/cat_backgrounds.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-size: cover;
}

.fashion {
  background: url(../Resources/Images/cat_fashion.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.nature {
  background: url(../Resources/Images/cat_nature.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.science {
  background: url(../Resources/Images/cat_science.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.education {
  background: url(../Resources/Images/cat_education.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.feelings {
  background: url(../Resources/Images/cat_feelings.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.health {
  background: url(../Resources/Images/cat_health.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.people {
  background: url(../Resources/Images/cat_people.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.religion {
  background: url(../Resources/Images/cat_religion.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.places {
  background: url(../Resources/Images/cat_places.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.animals {
  background: url(../Resources/Images/cat_animals.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.industry {
  background: url(../Resources/Images/cat_industry.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.computer {
  background: url(../Resources/Images/cat_computer.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.food {
  background: url(../Resources/Images/cat_food.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.sports {
  background: url(../Resources/Images/cat_sports.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.transportation {
  background: url(../Resources/Images/cat_transportation.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.travel {
  background: url(../Resources/Images/cat_travel.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.buildings {
  background: url(../Resources/Images/cat_buildings.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

.music {
  background: url(../Resources/Images/cat_music.jpg);
  transition: all 1s ease-in-out;
  cursor: pointer;
  background-position: center;
  background-size: cover;
}

/* end -style categories */

.categoryLink:hover {
  filter: grayscale(20%);
  background-size: cover;
}

/* Apply filer to images */

.filtered {
  background: rgb(236, 76, 138);
  position: relative;
}

img {
  display: block;
  max-width: 100%;
}

.filtered img {
  -webkit-filter: grayscale(100%);
  filter: grayscale(100%);
  opacity: 0.5;
}

.filtered,
.not-filtered {
  float: left;
  max-width: 50%;
  border: 10px solid white;
}

* {
  box-sizing: border-box;
}

/* DESKTOP */

@media (min-width: 768px) {
  #root {
    width: 80%;
    margin: auto;
  }

  header {
    height: 20vh;
    width: 100%;
    max-width: 100vw;
    overflow-x: hidden;
  }

  #headerImage {
    /* height: 100%; */
    background: url(../Resources/Images/header2.jpg);
    background-size: cover;

    /*
    background-position-x: left;
    background-position-y: 80%;
    */

    /* object-fit: content; */
    height: auto;
    max-width: 100%;
  }

  #headerText {
    font-size: 0.5rem;
  }

  #headerDescription {
    font-size: 1.3rem;
  }

  #categoryGrid {
    width: 100%;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: 0;
    margin-top: 24px;
    margin-bottom: 24px;
  }

  .categoryLink {
    font-size: 20px;
  }

  #desktopGridTitle {
    display: block;
    margin-top: 3%;
  }

  hr {
    display: block;
    margin: 12px;
    width: 15%;
  }

  .categoryContainer {
    /* height: 40vh; */
    border: none;
    display: flex;
    align-items: flex-end;
    padding: 20px;
  }

  #discoverAction {
    display: block;
    margin-top: 5%;
    transition: all 0.5s ease-in-out;
  }

  #discoverAction:hover {
    margin-top: 6%;
    background-color: red;
    width: 30px;
    height: 30px;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 100%;
    cursor: pointer;
  }
}
