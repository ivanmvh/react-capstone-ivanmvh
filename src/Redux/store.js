import { combineReducers, configureStore } from '@reduxjs/toolkit';
import photosSlice from './PhotosSlice';

const photosReducers = combineReducers({
  Photos: photosSlice,
});

const store = configureStore({
  reducer: photosReducers,
});

export default store;
