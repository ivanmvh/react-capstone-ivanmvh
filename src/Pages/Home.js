import { NavLink } from 'react-router-dom';
import { VscFoldDown } from 'react-icons/vsc';
import { useDispatch } from 'react-redux';
import { FaArrowRight } from 'react-icons/fa';
import categories from './categoriesList';
import { filterCategory } from '../Redux/PhotosSlice';
import './Home.css';

const Home = () => {
  const dispatch = useDispatch();

  const handleClick = (code, page, name) => {
    const parameters = {
      categoryCode: code,
      page,
      categoryName: name,
    };
    dispatch(filterCategory(parameters));
  };

  return (
    <div id="homeContainer">
      <header>
        <div id="headerImage" className="filtered">
          <img src="../Resources/Images/header2.jpg" alt="header" />
        </div>
        <div id="headerText">
          <h1>WELCOME TO PHOTO STORE</h1>
          <p id="headerDescription">A place where you can find the best photos for free</p>
          <a
            id="discoverAction"
            href="#categoryGrid"
          >
            <VscFoldDown style={{ color: 'white' }} />
          </a>
        </div>
      </header>
      <h2 id="gridTitle">Categories</h2>
      <div id="categoryGrid">
        {categories.map((category) => (
          <div
            className="categoryContainer"
            key={category.categoryCode}
          >
            <NavLink
              to={`./category/${category.categoryCode}`}
              key={category.categoryCode}
              onClick={() => handleClick(
                category.categoryCode,
                category.page,
                category.categoryName,
              )}
              className={`categoryLink ${category.categoryCode}`}
            >
              <div className="icon-wrapper">
                <FaArrowRight />
              </div>
              <h3 className="categoryTitle">{category.categoryName}</h3>
            </NavLink>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Home;
