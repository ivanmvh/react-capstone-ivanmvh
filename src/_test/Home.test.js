import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../Pages/Home';
import '@testing-library/jest-dom';
import store from '../Redux/store';

describe('Home', () => {
  const tree = test('renders Home component', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
