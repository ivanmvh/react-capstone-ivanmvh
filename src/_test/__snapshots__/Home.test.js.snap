// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Home component 1`] = `undefined`;

exports[`Navbar renders Navbar component 1`] = `undefined`;
